source/solution_deploy/source/wait_provider.py

```diff
@@ -0,0 +1,59 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Custom resource provider that waits for a specified time, then returns success"""
+
+from os import getenv
+import json
+from logging import basicConfig, getLevelName, getLogger
+from time import sleep
+import cfnresponse
+
+basicConfig(level=getLevelName(getenv("LOG_LEVEL", "WARNING")))
+logger = getLogger(__name__)
+
+
+class InvalidRequest(Exception):
+    """Invalid wait request"""
+
+
+def wait_seconds(wait: float):
+    """Wait for `wait` seconds"""
+    sleep(wait)
+
+
+def lambda_handler(event, context):
+    """Handle the Lambda request for a wait"""
+    response_data = {}
+
+    try:
+        properties = event.get("ResourceProperties", {})
+        logger.info(json.dumps(properties))
+
+        wait_create = float(properties["CreateIntervalSeconds"])
```

Comment:
Regarding placing variables close to their use, I think the scale here is small enough to not matter. I am taking the approach of validating input parameters first, then doing operations, rather than spreading the validation around.

I intentionally made the choice to not accept events that don't specify all the required properties. This reduces the complexity of the implementation and the amount of testing needed.

Reply:
when this was a conscious decision, I'm fine